server/superdesk/io/commands/update_ingest.py

```diff
@@ -151,7 +151,19 @@ class UpdateIngest(superdesk.Command):
         superdesk.Option('--provider', '-p', dest='provider_type'),
     )
 
+    @classmethod
+    def instantiate_if_class(cls, thing):
+        """Return an instance of `thing` if `thing` is a class, otherwise
+        return the `thing` itself.
+        """
+        return thing() if isinstance(thing, type) else thing
+
     def run(self, provider_type=None):
+        # instantiate any registered providers classes that have not been
+        # instantiated yet
+        for name, provider in providers.items():
+            providers[name] = self.instantiate_if_class(provider)
```

Comment:
What's the best place to instantiate the registered provider classes? To me this does not seem like the best place to do it. Is there some kind of an application "init()" that is run when the application is started?